contrib/auto-render/auto-render.js

```diff
@@ -25,10 +25,11 @@ const renderMathInText = function(text, delimiters) {
         } else {
             const span = document.createElement("span");
             const math = data[i].data;
+            // Override any display mode defined in the settings with that
+            // defined by the text itself
+            renderOptions.displayMode = data[i].display;
```

Comment:
The fact that you are modifying the agument passed to this function might lead to ugly surprises in some rare situations. Using `extend` to create a clone with one modification would have been a nice alternative, so I wonder whether it might be better to keep that function around.